src/main/java/jenkins/branch/MultiBranchProject.java

```diff
@@ -1870,12 +1864,16 @@ public void run() {
                 if (!scheduleBuilds.isEmpty()) {
                     listener.getLogger().println("Scheduling builds for branches:");
                     for (Map.Entry<P, SCMRevision> entry : scheduleBuilds.entrySet()) {
-                        listener.getLogger().println("    " + factory.getBranch(entry.getKey()).getName());
-                        if (entry.getKey().scheduleBuild(0, new SCMTrigger.SCMTriggerCause("Branch indexing"))) {
-                            try {
-                                factory.setRevisionHash(entry.getKey(), entry.getValue());
-                            } catch (IOException e) {
-                                e.printStackTrace(listener.error("Could not update last revision hash"));
+                        final P _project = entry.getKey();
+                        listener.getLogger().println("    " + factory.getBranch(_project).getName());
+                        if (_project instanceof ParameterizedJobMixIn.ParameterizedJob) {
+                            // TODO need a static helper method to schedule a ParameterizedJob
+                            if (new ParameterizedJobMixIn() {@Override protected Job asJob() {return _project;}}.scheduleBuild(0, new SCMTrigger.SCMTriggerCause("Branch indexing"))) {
```

Comment:
It's difficult to see this long lines in GH for review. It should be split, IMO.